go/mysql/binlog_event_compression.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysql
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/klauspost/compress/zstd"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// This file contains code related to handling compression related
+// events. More specifically today, compressed transaction payloads:
+// See: https://dev.mysql.com/doc/refman/en/binary-log-transaction-compression.html
+
+// Transaction Payload wire protocol fields:
+// https://dev.mysql.com/doc/dev/mysql-server/latest/classbinary__log_1_1codecs_1_1binary_1_1Transaction__payload.html
+const (
+	payloadHeaderEndMark = iota
+	payloadSizeField
+	payloadCompressionTypeField
+	payloadUncompressedSizeField
+)
+
+// Compression algorithms that are supported (only zstd today
+// in MySQL 8.0):
+// https://dev.mysql.com/doc/refman/8.0/en/binary-log-transaction-compression.html
+const (
+	transactionPayloadCompressionZstd = 0
+	transactionPayloadCompressionNone = 255
+)
+
+var transactionPayloadCompressionTypes = map[uint64]string{
+	transactionPayloadCompressionZstd: "ZSTD",
+	transactionPayloadCompressionNone: "NONE",
+}
```

Comment:
I might have a need for these to be public to; but I will iterate in a different PR if needed.